src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2360,19 +2359,19 @@ private void EmitAssignmentValue(BoundAssignmentOperator assignmentOperator)
             else
             {
                 int exprTempsBefore = _expressionTemps?.Count ?? 0;
-                var local = ((BoundLocal)assignmentOperator.Left).LocalSymbol;
+                var lhs = assignmentOperator.Left;
 
                 // NOTE: passing "ReadOnlyStrict" here. 
                 //       we should not get an address of a copy if at all possible
-                LocalDefinition temp = EmitAddress(assignmentOperator.Right, local.RefKind == RefKind.RefReadOnly ? AddressKind.ReadOnlyStrict : AddressKind.Writeable);
+                LocalDefinition temp = EmitAddress(assignmentOperator.Right, lhs.GetRefKind() == RefKind.RefReadOnly ? AddressKind.ReadOnlyStrict : AddressKind.Writeable);
 
                 // Generally taking a ref for the purpose of ref assignment should not be done on homeless values
                 // however, there are very rare cases when we need to get a ref off a temp in synthetic code.
                 // Retain those temps for the extent of the encompassing expression.
                 AddExpressionTemp(temp);
 
                 // are we, by the way, ref-assigning to something that lives longer than encompassing expression?
-                if (local.SynthesizedKind.IsLongLived())
+                if (assignmentOperator.Left.Kind == BoundKind.Local && ((BoundLocal)lhs).LocalSymbol.SynthesizedKind.IsLongLived())
```

Comment:
This will need to handle ref-assignments to parameters as well. They are all "long lived" and can be readonly too.
Could be done in a separate change.

Reply:
I tried adding support for parameters when I touched this last time. 
Then it required some refactoring and then I realized that I could not test the path for parameters anyways, so I reverted changes.

---
In reply to: [154417084](https://github.com/dotnet/roslyn/pull/23509#discussion_r154417084) [](ancestors = 154417084)

Reply:
@VSadov I don't think it's possible to hit this, even with my changes. Let's say there's an expression temp created during the assignment. If so, that temp has scope local to the method. The resulting expression must also then be scoped local to the method, meaning it can't be assigned to a parameter.

Am I missing a case?